diff --git a/nixos/modules/services/web-apps/mediawiki.nix b/nixos/modules/services/web-apps/mediawiki.nix
index f78880be6549..586ca1768e25 100644
--- a/nixos/modules/services/web-apps/mediawiki.nix
+++ b/nixos/modules/services/web-apps/mediawiki.nix
@@ -48,16 +48,25 @@ let
     preferLocalBuild = true;
   } ''
     mkdir -p $out/bin
-    for i in changePassword createAndPromote deleteUserEmail resetUserEmail userOptions edit nukePage update; do
-      echo "#! ${pkgs.runtimeShell}
+    makeWrapper ${php}/bin/php $out/bin/mediawiki-maintenance \
+      --set MEDIAWIKI_CONFIG ${mediawikiConfig} \
+      --add-flags ${pkg}/share/mediawiki/maintenance/run.php
+    for i in changePassword createAndPromote deleteUserEmail resetUserEmail userOptions edit nukePage update importDump run; do
+      script="$out/bin/mediawiki-$i"
+    sed -E 's/^ {2}//' <<'EOF' >"$script"
+      #!${pkgs.runtimeShell}
       become=(exec)
-      if [[ \"\$(id -u)\" != ${user} ]]; then
+      if [[ "$(id -u)" != ${user} ]]; then
         become=(exec /run/wrappers/bin/sudo -u ${user} --)
       fi
-      \"\''${become[@]}\" ${php}/bin/php ${pkg}/share/mediawiki/maintenance/run.php --conf ${mediawikiConfig} \
-        ${pkg}/share/mediawiki/maintenance/$i.php \"\$@\"
-      " > $out/bin/mediawiki-$i
-      chmod +x $out/bin/mediawiki-$i
+      "${"$"}{become[@]}" ${placeholder "out"}/bin/mediawiki-maintenance \
+    EOF
+      if [[ "$i" != "run" ]]; then
+        echo "  ${pkg}/share/mediawiki/maintenance/$i.php \"\$@\"" >>"$script"
+      else
+        echo "  ${pkg}/share/mediawiki/maintenance/\$1.php \"\''${@:2}\"" >>"$script"
+      fi
+      chmod +x "$script"
     done
   '';
