diff --git a/src/main.rs b/src/main.rs
index c804d16..fc0d1ea 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -52,18 +52,8 @@ struct Config {
 static CONFIG: LazyLock<Config> = LazyLock::new(Config::from_args);
 
 static GITHUB_TOKEN: LazyLock<OsString> = LazyLock::new(|| {
-    use std::io::{stdin, BufRead, BufReader};
-    use std::os::unix::prelude::*;
-
-    let mut bytes = Vec::with_capacity(41);
-    if let Err(e) = BufReader::new(stdin()).read_until(b'\n', &mut bytes) {
-        eprintln!("pr-tracker: read: {}", e);
-        exit(74)
-    }
-    if bytes.last() == Some(&b'\n') {
-        bytes.pop();
-    }
-    OsString::from_vec(bytes)
+    use std::env;
+    env::var_os("PR_TRACKER_GITHUB_TOKEN").unwrap()
 });
 
 #[derive(Debug, Default, Template)]
